meson.build

```diff
@@ -8,8 +8,39 @@ project('abseil-cpp', 'cpp',
 cpp = meson.get_compiler('cpp')
 
 cpp_flags = [
-  '-Wno-sign-compare'
+  '-Wcast-qual',
+  '-Wconversion-null',
+  '-Wformat',
+  '-Wformat-security',
+  '-Wmissing-declarations',
+  '-Woverlength-strings',
+  '-Wpointer-arith',
+  '-Wundef',
+  '-Wunused-local-typedefs',
+  '-Wunused-result',
+  '-Wvarargs',
+  '-Wvla',
+  '-Wwrite-strings',
+  '-DNOMINMAX',
 ]
+
+size_of_void_p_res = cpp.run('''#include <stdio.h>
+                      int main() {
+                         void *ptr;
+                         printf("%d", sizeof(ptr));
+                         return 0;
+                      }
+                      ''', name : 'size_of_void_p check')
+if build_machine.cpu() in ['x86_64', 'amd64', 'AMD64']
```

Comment:
This should use `cpu_family` instead, this use case is exactly what it was designed for.